lib/publiccloud/instance.pm

```diff
@@ -477,4 +481,166 @@ sub network_speed_test() {
     record_info("curl $rmt_host", $self->run_ssh_command(cmd => "curl -w '$write_out' -o /dev/null -v https://$rmt_host/", proceed_on_failure => 1));
 }
 
+=head2 measure_boottime
+
+    measure_boottime();
+
+Perfomrance measurement of the system Boot time. 
+Mainly used C<systemd-analyze> command for the data extraction.
+Data is then collected in an internal record, ready for storing in a DB.
+Set PUBLIC_CLOUD_PERF_COLLECT true or >0, to activate boottime measurements.
+
+=cut
+
+sub measure_boottime() {
+    my ($self, $instance, $type) = @_;
+    return 0 unless (get_var('PUBLIC_CLOUD_PERF_COLLECT'));
+
+    my $ret = {
+        kernel_release => undef,
+        kernel_version => undef,
+        type => undef,
+        analyze => {},
+        blame => {},
+    };
+
+    record_info("BOOT TIME", 'systemd_analyze');
+    # first deployment analysis
+    my ($systemd_analyze, $systemd_blame) = do_systemd_analyze_time($instance);
+    $ret->{analyze}->{$_} = $systemd_analyze->{$_} foreach (keys(%{$systemd_analyze}));
+    $ret->{blame} = $systemd_blame;
+    $ret->{type} = $type;
+    # $ret->{analyze}->{ssh_access} = $startup_time; # placeholder for next implementation
+
+    # Collect kernel version
+    $ret->{kernel_release} = $instance->run_ssh_command(cmd => 'uname -r', proceed_on_failure => 1);
+    $ret->{kernel_version} = $instance->run_ssh_command(cmd => 'uname -v', proceed_on_failure => 1);
+
+    # Do logging to openqa UI
+    $Data::Dumper::Sortkeys = 1;
+    record_info("RESULTS", Dumper($ret));
+
+    my $logs = '/var/log/cloudregister /var/log/cloud-init.log /var/log/cloud-init-output.log /var/log/messages /var/log/NetworkManager';
+    $instance->run_ssh_command(cmd => 'sudo tar -czvf /tmp/sle_cloud.tar.gz ' . $logs, proceed_on_failure => 1);
+    $instance->upload_log('/tmp/sle_cloud.tar.gz', failok => 1);
+
+    return $ret;
+}
+
+
+=head2 store_boottime_db
+
+    store_boottime_db();
+
+Save data collected with measure_boottime in a DB;
+Mainly stored on a remote InfluxDB on a Grafana server.
+
+=cut
+
+sub store_boottime_db() {
+    my ($self, $results) = @_;
+    return unless (get_var('_PUBLIC_CLOUD_PERF_PUSH_DATA') && $results);
+
+    my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI');
```

Comment:
```
my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI');
my $token = get_var('_PUBLIC_CLOUD_PERF_DB_TOKEN');
unless ($url && $token) {
        record_info("WARN", "PUBLIC_CLOUD_PERF_DB_URI or _PUBLIC_CLOUD_PERF_DB_TOKEN is missing ", result => 'fail');    
}
```

If _PUBLIC_CLOUD_PERF_PUSH_DATA is defined but PUBLIC_CLOUD_PERF_DB_URI or _PUBLIC_CLOUD_PERF_DB_TOKEN is missing I would assume some misconfiguration in such case . We still don't want to fail but we can add something which will drug attention of reviewer ...

Reply:
> We still don't want to fail but

But if we cannot write to any DB_URI, could we set it as soft failure, at least ?

Reply:
no,  we may confuse a lot of test results in such case , let's keep it transparent for other tests . "keep transparent" in such case means no job status change and minimizing chances for test failure  as it will be not the main goal of any test which we starting

Reply:
ah sorry , I forgot to add return ofc. it should be 
```
my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI');
my $token = get_var('_PUBLIC_CLOUD_PERF_DB_TOKEN');
unless ($url && $token) {
        record_info("WARN", "PUBLIC_CLOUD_PERF_DB_URI or _PUBLIC_CLOUD_PERF_DB_TOKEN is missing ", result => 'fail');    
       return -1;
}